core/src/main/java/org/apache/brooklyn/core/catalog/CatalogPredicates.java

```diff
@@ -39,21 +39,65 @@
 public class CatalogPredicates {
 
     public static <T,SpecT> Predicate<CatalogItem<T,SpecT>> isCatalogItemType(final CatalogItemType ciType) {
-        return new Predicate<CatalogItem<T,SpecT>>() {
+        // TODO PERSISTENCE WORKAROUND kept anonymous function in case referenced in persisted state
```

Comment:
Are you saying to add this to the `private static class` that I added, which this now uses? Not sure how `@since 0.8.0` relates to this given line?